nevermined-io/sdk-js
# Migrating DDOHelper functions to DDO

- Aligning with NGINX poc
- linting
- updating the authentication token
- 1d
- parametrization of the service
- feat: bumping to v1.0.0
- feat: Adding v1.0.0 Changelog updates
- fix: renaming `master` branch to `main`
- fix: renaming to `main` branch and using pinned version of `node`
- ci: fix tests
- ci: using `main` branch
- ci: testing with https services
- enabling test
- Feature/subscriptions (#474)
- feat: added method to retrieve all subscription DDOs created by address
- feat: added method to get all subscription DDOs purchased
- feat: method to search for all services belonging to a subscription
- fix: testing node workflow
- fix: unskip test
- feat: bump version to 1.2.0
- find datasets associated to a subscription
- update sdk version
- update nvm-actions 0.8.0
- feat: adding subscription asset type
- test: subscription type
- feat: adding v1.3.1 Changelog updates
- feat: adds utility functions to manage unlisting and voting
- feat: renaming method for clarity
- ci: fixing DDO tests because curation attributes
- ci: default node version
- feat: adding v1.3.2 Changelog updates
- ci: integration with proxy after refactor
- ci: host header is not necessary from the proxy side
- fix: only update did on-chain when using metadata on immutable store
- feat: removing wallet interaction (sign DDO) because don't provide much value here
- fix: getting ownership from on-chain
- feat: promoting marketplace api
- feat: adding v1.3.3 Changelog updates
- ci: using latest node
- ci: adding tests
- ci: adding more nft 721 balance tests
- test: improve dataset subscription tests
- ci: test failing
- fix: use expiration on the nft sales template
- ci: adding `getMintedEntries` function
- feat: add hyperspace to network names
- test: add subscription duration tests
- test: added subscription duration tests
- test: added more subscription duration tests
- chore: bump version to 1.3.4-rc0
- test: fix tests
- Automated CHANGELOG.md update
- adding a case for dleq dtp
- types for dleq
- ci: using `latest` contracts
- ci: using latest abi
- test: fixing tests
- test: excluding templateId of validation
- Adding v1.3.0 Changelog updates
- add id in provenance result
- Automated CHANGELOG.md update
- fix: issue with the escrow payment condition subgraph
- chore: lint
- fix: testing node should use contracts 3.1.0
- Automated CHANGELOG.md update
- fix: using nevermined.app domain
- fix: using contracts `v3.1.0` for testing
- Create dependabot.yaml
- fix: extension
- fix: upgrade action to v0.10.0
- fix: update version in nightly run
- feat: test with 3.2.1
- fix: node should run with NO_GRAPH flag set to true
- feat: use node 1.2.5
- chore(deps-dev): bump auto-changelog from 1.16.4 to 2.4.0
- chore(deps-dev): bump @types/temp from 0.8.34 to 0.9.1
- chore(deps-dev): bump typescript from 4.9.5 to 5.0.4
- chore(deps-dev): bump webpack-cli from 4.10.0 to 5.0.2
- chore(deps-dev): bump cross-env from 6.0.3 to 7.0.3
- chore(deps-dev): bump @types/node from 12.20.55 to 20.1.0
- feat: adding Gnosis networks
- feat: adding v1.3.7 Changelog updates
- update-pkg script working
- bump version
- Automated CHANGELOG.md update
- feat: allowing to define timelock and timeout parameters
- ci: removing not needed variable
- ci: fix problem with timelock and timeout assignations
- feat: bumping package.json to v1.3.9
- feat: adding v1.3.9 Changelog updates
- use promise settled instead of promise all
- feat: sdk working in offchain mode
- fix: message error
- fix: avoid update config object
- feat: adding the optional attribute openUrls to the web services
- fix: address comments
- fix: address more comments
- Automated CHANGELOG.md update
- feat: expose search by did and nft address utility function
- ci: documentation and throwing exception
- ci: wait elastic to index documents
- fix: setting the right version
- fix: setting the right version v1.3.11
- feat: adding v1.3.11 Changelog updates
- feat: refactor get webservices and datasets by subscription
- Update testing.yml
- fix: upgrade action to v0.11.0
- fix: try with both variables
- fix: udpate config
- feat: setting up a default fulfill timeout
- feat: bumping to v1.3.12
- ci: fixing timeout test
- feat: adding v1.3.12 Changelog updates
- chore: fix small non-breaking issues
- fix: sending did for transfer nft through Node
- chore: bumping to v1.3.13
- chore: removing console
- chore: addressing comments
- ci: using compute template
- feat: improve getAgreement
- chore: refactor fetching agreement data
- chore(deps-dev): bump ora from 4.1.1 to 6.3.1
- chore(deps-dev): bump typedoc from 0.23.28 to 0.24.7
- Automated CHANGELOG.md update
- fix: wrong function call
- chore: version bump
- feat: adding v1.3.14 Changelog updates
- fix: events once method should not return unless there are events
- fix: integration tests
- fix: bookmark test
- Automated CHANGELOG.md update
- fix: adding tokenSupply to erc20
- fix: using token from config address
- chore: initializing default contract address
- feat: adding v1.13.16 Changelog updates
- fix: use gas station on polygon networks
- feat: calculate fees when using contactHandler.deployAbi
- fix: set manual gas limit to 10000000 when deploying contract
- feat: refactor to use new nevermined.utils.contractHandler
- chore: bump version to 1.4.0
- chore: remove unused methods
- Automated CHANGELOG.md update
- feat: add custom nested queries to search methods
- feat: update sdk version
- Automated CHANGELOG.md update
- feat: encoded DID using base36 format
- feat: integration of open endpoints
- chore: bumping to v1.4.0-rc0
- chore: adding v1.4.0-rc0 Changelog updates
- chore: fixing version
- chore: adding v1.4.1-rc0 Changelog updates
- chore: v1.4.2-rc0
- chore: adding v1.4.2-rc0 Changelog updates
- chore: updating npm publishing workflow
- chore: publishing v1.4.2-rc1
- chore: adding v1.4.2-rc1 Changelog updates
- test: adding open endpoints
- chore: workflow cleanup
- chore: adding v1.4.2 Changelog updates
- fix: handling encoding problem with dids starting by 0
- ci: fixed version of opengsn dep
- chore: adding v1.4.3 Changelog updates
- ci: using same version as opengsn docker image
- ci: skipping metatx test
- ci: skipping metatx test
- feat: adding networkId where the asset is published
- chore: adding v1.4.4 Changelog updates
- fix: delete assets
- fix: test
- feat: use actions with elastic 8
- feat: bump version
- feat: bump version
- Automated CHANGELOG.md update
- feat: reduce the number of on chain calls required to get and agreement status
- Automated CHANGELOG.md update
- fix: update polygon gasstation urls
- Automated CHANGELOG.md update
- feat: allow jwt tokens with eip-712 typed signatures
- chore: bump version to 1.5.3
- chore: adding basic authentication testing options
- chore: linting
- Automated CHANGELOG.md update
- chore: remove wrong line
- chore: skipping test
- fix: change resolve by retrieveDDO in servicesBySubscription search method
- feat: keep testing compute in nightly
- Automated CHANGELOG.md update
- refactor: update to latest subgraphs
- feat: adding helper methods to check if a service exists
- chore: migrating DDOHelper functions to DDO

## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes


## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes

#### Funny gif

![Put a link of a funny gif inside the parenthesis-->]()
